### DISCUSSION TITLE
Static website with dynamic component

### DISCUSSION THREAD

**Discussion by renatocassino** [vercel/next.js#24325] [Help]
Hello everyone

I have an website with SSR and an CMS to serve the data. I'm trying to migrate to a static website.

I implemented the getStaticProps and getStaticPaths to my dynamic routes, but I have some components that must run on client-side

The first example is my search with autocomplete. When I build my website the search doesn't work anymore. The second is a list of the latest articles visited (stored in localStorage). This component doesn't work too

I cannot find anything about this, but I understand that when the static webpage is built all javascript related to react stop working, I'm right?

Is possible to put these two components running in client-side and all other pages in static?

Thanks for this

**Comment by akellbl4** [comment]
Hi
When you build Next.js as SSG it keeps React and all other JavaScript for client-side rendering.
Could you please tell me more about the problems?

**Comment by bobarros** [comment]
Only js that needs to access the server's not going to work. 
Searching features it's a good example.

For pages that need to send a request to serve, use getServerSideProps.
